app/controllers/actions/JourneyAction.scala

```diff
@@ -36,13 +37,18 @@ class JourneyAction @Inject()(cacheService: ExportsCacheService)(
     implicit val hc: HeaderCarrier =
       HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
 
-    request.declarationId match {
-      case Some(id) =>
-        cacheService.get(id).map {
-          case Some(declaration) => Right(JourneyRequest(request, declaration))
-          case _                 => handleMissingDeclarationId(request)
+    Mode.binder.bind("mode", request.queryString) match {
+      case Some(Right(mode)) =>
+        request.declarationId match {
+          case Some(id) =>
+            cacheService.get(id).map {
+              case Some(declaration) => Right(JourneyRequest(request, declaration, mode))
+              case _                 => handleMissingDeclarationId(request)
+            }
+          case None => Future.successful(handleMissingDeclarationId(request))
```

Comment:
I'll at least combine those two options with `handleMissingDeclarationId` to one, but it's old code

Reply:
You know that one is before database hit and second is after ?

Reply:
yeah, you're right